### PR TITLE
scrcpy: 1.3 -> 1.5

### PR DESCRIPTION
**Pull request by tomberek**
###### Motivation for this change
Update to newest version

###### Things done
Updated version.
Added configurations to allow hardware and software rendering.
Added configuration to remove unneeded errors.
Added configuration to use prebuilt server directly.
Tested on Ubuntu, NixOS.

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/misc/scrcpy/default.nix [hunk 1]
```diff
@@ -22,8 +22,14 @@ stdenv.mkDerivation rec {
...
     repo = "scrcpy";
     rev = "v${version}";
-    sha256 = "02szi8w3w0lacyz42hlayxififi863qpm63yg9qir3jcl2vs7vdk";
+    sha256 = "015j1h161c8s2vyh5rbpj0q473zbp5sygw5y7cyv59237d703kcb";
   };
+  postPatch = ''
+    substituteInPlace app/src/screen.c \
+      --replace "SDL_RENDERER_ACCELERATED" "SDL_RENDERER_ACCELERATED || SDL_RENDERER_SOFTWARE"
+    substituteInPlace app/src/command.c \
+      --replace 'const char *const adb_cmd[] = {"shell", "rm", path};' 'const char *const adb_cmd[] = {"shell", "rm", "-f", path};'
```
Comment (Mic92) [inline pkgs/misc/scrcpy/default.nix]: Why is this necessary for nixpkgs and not by default?

File: pkgs/misc/scrcpy/default.nix [hunk 2]
```diff
@@ -22,8 +22,14 @@ stdenv.mkDerivation rec {
     owner = "Genymobile";
     repo = "scrcpy";
     rev = "v${version}";
-    sha256 = "02szi8w3w0lacyz42hlayxififi863qpm63yg9qir3jcl2vs7vdk";
+    sha256 = "015j1h161c8s2vyh5rbpj0q473zbp5sygw5y7cyv59237d703kcb";
   };
+  postPatch = ''
+    substituteInPlace app/src/screen.c \
+      --replace "SDL_RENDERER_ACCELERATED" "SDL_RENDERER_ACCELERATED || SDL_RENDERER_SOFTWARE"
```
Comment (Mic92) [inline pkgs/misc/scrcpy/default.nix]: What is the effect of this patch?

File: pkgs/misc/scrcpy/default.nix (hunk 1 above)
Comment (tomberek) [reply pkgs/misc/scrcpy/default.nix]: When copying over the prebuilt binary, it also copies the permissions of the nix store, and thus it cannot delete normally.

File: pkgs/misc/scrcpy/default.nix (hunk 2 above)
Comment (tomberek) [reply pkgs/misc/scrcpy/default.nix]: When run without a hardware accelerator, this allows the command to continue working rather than failing unexpectedly. This can happen when running on non-NixOS because then scrcpy seems to have a hard time using the host OpenGL-supporting hardware. It would be better to fix the OpenGL problem, but that seems much more intrusive.

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Can you add comments with the explanations you've given to @Mic92?

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build scrcpy

**Comment by worldofpeace** [comment]
Shouldn't we be at this [version](https://github.com/Genymobile/scrcpy/releases/tag/v1.5-fixversion)?

**Comment by tomberek** [comment]
@worldofpeace good catch!

File: pkgs/misc/scrcpy/default.nix [hunk 3]
```diff
@@ -22,9 +22,22 @@ stdenv.mkDerivation rec {
     owner = "Genymobile";
     repo = "scrcpy";
     rev = "v${version}";
```
Comment (worldofpeace) [inline pkgs/misc/scrcpy/default.nix]: I think we need to use the `fixversion` tag here also.
